File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/render/EffectUpdater.java
package com.abubusoft.xenon.render;

import com.abubusoft.xenon.shader.Shader;

/**
 * <p></p>
 * @author Francesco Benincasa
 *
 * @param <P>
 * @param <E>
 */
public interface EffectUpdater<P extends AbstractEffect<E>, E extends Shader> {

	/**
	 * @param effect
	 * @param shader
	 * @param enlapsedTime
	 * @param speedAdapter
	 */
	public void update(P effect, E shader, long enlapsedTime, float speedAdapter);
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/render/EffectUpdater.kt
package com.abubusoft.xenon.render

import com.abubusoft.xenon.shader.Shader

/**
 *
 *
 * @author Francesco Benincasa
 *
 * @param <P>
 * @param <E>
</E></P> */
interface EffectUpdater<P : AbstractEffect<E>, E : Shader> {
    /**
     * @param effect
     * @param shader
     * @param enlapsedTime
     * @param speedAdapter
     */
    fun update(effect: P, shader: E, enlapsedTime: Long, speedAdapter: Float)
}